packages/botonic-nlu/src/preprocessor.ts

```diff
@@ -0,0 +1,119 @@
+/* eslint-disable @typescript-eslint/unbound-method */
+import { DefaultTokenizer } from './preprocessing-tools/tokenizer';
+import { DefaultStemmer } from './preprocessing-tools/stemmer';
+import { DefaultNormalizer } from './preprocessing-tools/normalizer';
+import { UNKNOWN_TOKEN } from './constants';
+import { DataSet, Vocabulary, Normalizer, Stemmer, Tokenizer } from './types';
+import { Language } from './language';
+
+export class Preprocessor {
+  private _normalizer: Normalizer;
+  private _stemmer: Stemmer;
+  private _tokenizer: Tokenizer;
+  private _language: Language;
```

Comment:
this should not compile with your tsconfig as _language (and others) will be undefined after constructor.
Try calling tsc